hypothesis-python/src/hypothesis/internal/conjecture/data.py

```diff
@@ -72,16 +72,13 @@ def wrapper(tp):
         return wrapper
 
 
-ONE_BOUND_INTEGERS_LABEL = calc_label_from_name("trying a one-bound int allowing 0")
-INTEGER_RANGE_DRAW_LABEL = calc_label_from_name("another draw in integer_range()")
-BIASED_COIN_LABEL = calc_label_from_name("biased_coin()")
-
 TOP_LABEL = calc_label_from_name("top")
 DRAW_BYTES_LABEL = calc_label_from_name("draw_bytes() in ConjectureData")
-DRAW_FLOAT_LABEL = calc_label_from_name("drawing a float")
-FLOAT_STRATEGY_DO_DRAW_LABEL = calc_label_from_name(
-    "getting another float in FloatStrategy"
-)
+DRAW_FLOAT_LABEL = calc_label_from_name("draw_float() in PrimitiveProvider")
+DRAW_INTEGER_LABEL = calc_label_from_name("draw_integer() in PrimitiveProvider")
+DRAW_STRING_LABEL = calc_label_from_name("draw_string() in PrimitiveProvider")
+DRAW_BYTES_LABEL = calc_label_from_name("draw_bytes() in PrimitiveProvider")
+DRAW_BOOLEAN_LABEL = calc_label_from_name("draw_boolean() in PrimitiveProvider")
```

Comment:
I've changed when examples/discards are invoked here, based on a potentially flawed understanding. As I understand it, examples are useful as atomic building blocks to do things like (1) generating mutations (`generate_mutations_from`) or (2) shrinking them as a single entity (e.g. `try_shrinking_blocks`). This sounds an awful lot like the reasoning behind our new ir! But examples can also contain multiple ir nodes that will be shrunk/mutated together.

Based on this understanding, I've done the following:
- Remove all sub-ir examples. Since these don't correspond to draws anymore, they don't have an effect (I hope).
- Add an example for each ir draw. This was already happening either explicitly or implicitly for ir nodes with existing helpers (floats, integers), potentially split across multiple labels for multiple cases, but this formalizes it, and adds it for new nodes (draw_string).
- Leave any other examples, at levels above the ir, alone. We may be able to revisit these and decide they don't have an effect now that we have ir node examples (or indeed have a negative effect due to effectively duplicating an ir node example).

Reply:
sub-IR examples could in principle do things, but I'd guess the effect is negligible, and we want to obsolete this anyway (e.g. by shrinking at the IR level too) so the change sounds good to me.

A neat aspect of the mutation/shrinking use of the labels is that we can tell when things are "similar" in a relevant way, e.g. likely to be swappable.  That to me implies that we might want to include the kwargs in the label, but I think we actually handle that by using the hash of the strategy object itself further up, so... no action needed?   Probably, anyway; we should do some benchmarks before we merge.

Reply:
I'm less familiar with the mutation code, but my intuition is that perhaps including the IR draw kwargs in the label hurts more than it helps, because it means that we'll only try to swap *identical* kwarg draws. But there are lots of differing kwargs that can be swapped with no issues, because they happen to draw the same sequence of bits: e.g. `draw_integer(0, 200)` and `draw_integer(0, 201)`.

Reply:
Agreed that leaving it out of the labels makes sense, given how we use the labels.

I was thinking that we could in future check if a swap was valid by doing a check against their respective kwargs, without actually running (or simulating) the test.  That's for later though.

Reply:
yup, lots of cool potential using the kwargs as a backing!